src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestEventsConverter.kt

```diff
@@ -0,0 +1,203 @@
+package org.rust.cargo.runconfig.test
+
+import com.intellij.execution.process.ProcessOutputTypes
+import com.intellij.execution.testframework.TestConsoleProperties
+import com.intellij.execution.testframework.sm.ServiceMessageBuilder
+import com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter
+import com.intellij.openapi.util.Key
+import jetbrains.buildServer.messages.serviceMessages.ServiceMessage
+import jetbrains.buildServer.messages.serviceMessages.ServiceMessageVisitor
+
+/**
+ * Translate Cargo Test output to TeamCity Service Messages which SM Test Runner understands.
+ *
+ * Unfortunately, as for now Cargo Test only supports `human` message format, which is pretty
+ * obscure to handle programmatically. First of all, it is split into some separate sections,
+ * which are not easily distinguishable, that's why we do state machines here. Next problem is,
+ * that Cargo collects failed tests stdouts and print them yet at the end of test session.
+ * Lastly, we do not have exact information when given test started.
```

Comment:
I have no idea whether the thing I am doing here makes sense.

Cargo test output is terrible for machine parsing, especially test failures. This is fact. Currently I'm trying to collect failed tests, then collect their stdouts and print everything at the end of the test session (just like TeamCity plugin does). This has the advantage that we bind tests with their outputs, but we do not have information about them until end of run, also it complicates thing drastically.

Otherwise we can put this PR on hold, and do some cargo refactorings to support SM or other more machine-friendly format. I can try to do it, but I'm short on spare time and haven't made any contribution to Rust core projects, so this would take a long time to complete.